dotCMS/src/main/java/com/dotmarketing/filters/VanityUrlRequestWrapper.java

```diff
@@ -0,0 +1,101 @@
+package com.dotmarketing.filters;
+
+import static com.dotmarketing.filters.Constants.CMS_FILTER_QUERY_STRING_OVERRIDE;
+import static com.dotmarketing.filters.Constants.CMS_FILTER_URI_OVERRIDE;
+import java.nio.charset.Charset;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URLEncodedUtils;
+import com.dotcms.vanityurl.model.VanityUrlResult;
+import com.dotmarketing.util.UtilMethods;
+import com.google.common.collect.ImmutableMap;
+
+
+/**
+ * The VanityUrlOverrideRequest merges the parameters set in the original request and merges them
+ * with the parameters that are being set in the vanityUrl. In all cases, if there are parameters
+ * set in the VanityURL Query String, they will override the ones being sent in by original visitors
+ * request
+ *
+ */
+class VanityUrlRequestWrapper extends HttpServletRequestWrapper {
+    
+    final Map<String, String[]> queryParamMap;
+    final String newQueryString;
+    final boolean vanityHasQueryString;
+
+
+    public VanityUrlRequestWrapper(HttpServletRequest request, VanityUrlResult vanityUrlResult) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'vanityUrlResult' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5391667)